packages/doenetml-worker-rust/lib-doenetml-core/src/core/dependency.rs

```diff
@@ -348,6 +344,199 @@ pub fn create_dependencies_from_instruction_initialize_essential(
                 }
             }
 
+            if dependencies.is_empty() {
+                // Found no matching children.
+                // Create an essential dependency with the default_value for the state variable
+
+                // Treat the essential data as though it came from the first string child
+                // of the component, except recursing to extend source components
+                // in order to share the essential data with the extend source.
+
+                let source_idx =
+                    get_recursive_extend_source_component_when_exists(components, component_idx);
```

Comment:
I think in most cases, we just find the next extend source and don't keep recursing until we find the "original" source. Usually, we don't have to recurse because what we need is propagated each time a component extends something. However, we store essential data on the "original" extend source so that it is stored in just one place and then shared by all extenders. This function is just used to find where to create that essential data. (There isn't a one-level version of this exact function, but just similar functions the get the extend source.)

I'm not sure of a better name. Could say `get_original_extend_source_...`, though we haven't used that terminology `original` anywhere. `get_all` doesn't work, as it only gets a single one, i.e., the one that doesn't have another component extend source.

Reply:
What about `get_extend_source_origin`?

Reply:
I like it!